rclcpp/src/rclcpp/executors/multi_threaded_executor.cpp

```diff
@@ -103,7 +110,15 @@ MultiThreadedExecutor::run(size_t)
     execute_any_executable(any_exec);
```

Comment:
1. Thread-A has got Timer-X, and right before `execute_any_executable` (that said Timer-X is still alive) context switches.